web/packages/teleterm/src/ui/Search/pickers/ActionPicker.tsx

```diff
@@ -399,6 +397,44 @@ export function KubeItem(props: SearchResultItem<SearchResultKube>) {
   );
 }
 
+export function NoResultsItem(props: { clusters: tsh.Cluster[] }) {
+  const excludedClustersCopy = getExcludedClustersCopy(props.clusters);
+  return (
+    <NonInteractiveItem>
+      <Item Icon={icons.Info} iconColor="text.primary">
+        <Text typography="body1">No matching results found.</Text>
+        {excludedClustersCopy && (
+          <Text typography="body1" color="text.primary">
+            {excludedClustersCopy}
+          </Text>
+        )}
+      </Item>
+    </NonInteractiveItem>
+  );
+}
+
+export function TypeToSearchItem() {
+  return (
+    <NonInteractiveItem>
+      <Text typography="body1" color="text.primary">
+        Type something to search.
+      </Text>
+    </NonInteractiveItem>
+  );
+}
+
+function getExcludedClustersCopy(allClusters: tsh.Cluster[]): string {
+  const excludedClusters = allClusters.filter(c => !c.connected);
+  const excludedClustersString = excludedClusters.map(c => c.name).join(', ');
+  if (excludedClusters.length === 0) {
+    return '';
+  }
+  if (excludedClusters.length === 1) {
+    return `The cluster ${excludedClustersString} was excluded from the search because you are not logged in to it.`;
```

Comment:
The filters should be taken into account when preparing the copy. Here I filter by a specific cluster to which I'm logged in to but the copy still says that an unrelated cluster was not included in the search results.

![excluded clusters copy](https://user-images.githubusercontent.com/27113/231132555-a4f6a0b9-e5e0-48a9-8b81-b2b79d391d72.png)